zxlive/mainwindow.py

```diff
@@ -350,7 +356,47 @@ def reduce() -> None:
             cmd = AddRewriteStep(self.active_panel.graph_view, new_graph, self.active_panel.step_view, reduction["text"])
             self.active_panel.undo_stack.push(cmd)
         return reduce
-    
+
+    def create_new_rewrite(self) -> None:
```

Comment:
It might be nicer to move most of this logic to its own file, like how the other messageboxs are in the dialogs.py file (so maybe could just put it there for now?)